src/main/java/com/ea/restaurant/repository/UserRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.ea.restaurant.repository;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+  User findByIdWithStatus(Long id);
```

Comment:
This or not return a user, use `Optional<User>` instead.

Should be `findByIdAndStatus(String id, Status status)`

Reply:
Done.